src/glyph.rs

```diff
@@ -0,0 +1,156 @@
+use glib::translate::*;
+use pango_sys;
+use GlyphItem;
+use Item;
+use GlyphString;
+
+impl GlyphString {
+
+    pub fn num_glyphs(&self) -> i32 {
+        unsafe { (*self.to_glib_none().0).num_glyphs }
+    }
+
+    pub fn glyph_info(&self) -> Vec<GlyphInfo> {
+        if self.num_glyphs() < 0 {
+            return Vec::new()
+        }
+        let num_glyphs = self.num_glyphs() as usize;
+        unsafe {
+            let glyphs: *mut pango_sys::PangoGlyphInfo = (*self.to_glib_none().0).glyphs;
+            FromGlibContainer::from_glib_none_num(glyphs, num_glyphs)
+        }
+    }
+}
+
+impl GlyphItem {
+    pub fn item(&self) -> Item {
+        unsafe { from_glib_none((*self.to_glib_none().0).item) }
+    }
+
+    pub fn glyph_string(&self) -> GlyphString {
+        unsafe { from_glib_none((*self.to_glib_none().0).glyphs) }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct GlyphInfo(*mut pango_sys::PangoGlyphInfo);
+
+impl GlyphInfo {
+    pub fn glyph(&self) -> u32 {
+        unsafe { (*self.0).glyph }
+    }
+
+    pub fn geometry(&self) -> &GlyphGeometry {
+        unsafe { &*(&((*self.0).geometry) as *const _ as *const GlyphGeometry) }
+    }
+}
+
+impl FromGlibContainerAsVec<*mut pango_sys::PangoGlyphInfo, *mut pango_sys::PangoGlyphInfo> for GlyphInfo {
+    
+    unsafe fn from_glib_none_num_as_vec(ptr: *mut pango_sys::PangoGlyphInfo, num: usize) -> Vec<GlyphInfo> {
+        if num == 0 || ptr.is_null() {
+            return Vec::new()
+        }
+        let mut res = Vec::with_capacity(num);
+        for x in 0..num {
+            res.push(from_glib_none(ptr.add(x)));
+        }
+        res
+    }
+
+    unsafe fn from_glib_container_num_as_vec(_ptr: *mut pango_sys::PangoGlyphInfo, _num: usize) -> Vec<Self> {
+        unimplemented!();
```

Comment:
Maybe this function need be like https://github.com/gtk-rs/glib/blob/master/src/translate.rs#L1464.
from_glib_full_num_as_vec IMHO too but not sure better call from_glib_none_num_as_vec or make copy with `from_glib_none(ptr.add(x))` ?
@sdroege what you think?

Reply:
Thanks, I wasn't sure about this and was gonna ask. 
I made the changes as you suggested